### ISSUE TITLE
Connect to a logic without a market / connect to a market without a logic

### ISSUE THREAD

**Issue by jkrivine** [mangrovedao/mangrove#111]
Currently using the API on the maker side can only be done by connecting to an offer logic on a specific market using `mgv.MakerConnect({address, base, quote})`. 
However there are use cases where one wants to connect to a logic to read or set its mutable values. There are also use cases where one wants to market make using a simple wallet.
 
Maybe the following could be accessible:
* `const mkr = mgv.connectOfferLogic` to connect to a logic
* `const mkr = mgv.connectEOA` to connect to a wallet
* `const mkrAB = mkr.connectMarket({base:B,quote:A})` to connect the mkr API to a market
this implies that some functions of the mkr API should return an error if not connected to a market

**Comment by jkrivine** [comment]
An alternative is to have EOA makers using `market` functions to post liquidity. This would be consistent since `market` functions allow EOA to pass taker orders already. Then in `market` one would need to connect to a logic in order to post reactive offers. In summary I propose:
* `const market = mgv.market(...);`
* `market.bid(...)` and `market.ask(...)` for EOA makers
* `const logic = market.connectLogic(....)` to connnect to a specific offer logic
* `logic.bid(...)` and `logic.ask(...)` onchain makers

**Comment by espendk** [comment]
I rather like your second proposal.

One idea for avoiding forcing the user to connect to a market just for connecting to the logic could be something like:

```javascript
    const market = mgv.market(...);
    market.bid(...) / market.ask(...); // for EOA makers
    const logic = MyLogic.connect(...); // to connnect to a specific offer logic
    const logicOnMarket = logic.connectMarket(market); // to connect the logic to a specific market
    logicOnMarket.bid(...) / logicOnMarket.ask(...) // onchain makers
```

The type of `logicOnMarket` could be an extension of `logic` with market operations.

**Comment by jkrivine** [comment]
So `MyLogic` would be an independent class?

**Comment by jkrivine** [comment]
would make sense. It would essentially wrap ethers.js calls to functions of https://github.com/mangrovedao/mangrove/blob/master/packages/mangrove-solidity/contracts/Strategies/interfaces/IOfferLogic.sol

**Comment by espendk** [comment]
> So `MyLogic` would be an independent class?

Yes. I thought it already was, since you mentioned "access the offer logic". Also, that pattern could perhaps make it easier to plug in external offer logics ?

**Comment by jkrivine** [comment]
Yes agreed.

**Comment by jkrivine** [comment]
Addressed in commit #146
